scripts/cat/save_load.py

```diff
@@ -0,0 +1,151 @@
+import os
+from typing import TYPE_CHECKING, Type
+
+import ujson
+
+from scripts.game_structure.game.save_load import safe_save
+from scripts.game_structure.game.settings.settings import get_game_setting
+from scripts.housekeeping.datadir import get_save_dir
+
+if TYPE_CHECKING:
+    from scripts.cat.cats import Cat
+    from scripts.game_structure.game_essentials import Game
+
+faded_ids = []
+"""List of IDs of faded cats"""
+
+cat_to_fade = []
+"""Cats who have been faded since the last save"""
+
+
+def save_cats(clanname, cat_class: Type["Cat"], game: "Game"):
+    """Save the cat data."""
+
+    directory = get_save_dir() + "/" + clanname
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+    # Delete all existing relationship files
+    if not os.path.exists(directory + "/relationships"):
+        os.makedirs(directory + "/relationships")
+    for f in os.listdir(directory + "/relationships"):
+        os.remove(os.path.join(directory + "/relationships", f))
+
+    save_faded_cats(clanname, cat_class, game)  # Fades cat and saves them, if needed
+
+    clan_cats = []
+    for inter_cat in cat_class.all_cats.values():
+        cat_data = inter_cat.get_save_dict()
+        clan_cats.append(cat_data)
+
+        inter_cat.save_condition()
+
+        if inter_cat.history:
+            inter_cat.save_history(directory + "/history")
+            # after saving, dump the history info
+            inter_cat.history = None
+        if not inter_cat.dead:
+            inter_cat.save_relationship_of_cat(directory + "/relationships")
+
+    safe_save(f"{get_save_dir()}/{clanname}/clan_cats.json", clan_cats)
```

Comment:
Moving forward, we should try to use Pathlib instead of constructing path literals as strings.